bin/user/weatherservices.py

```diff
@@ -1431,183 +2052,580 @@ def getRecord(self):
+                dt = dateutil.parser.isoparse(obsval)
+                # convert dt timestamp to unix timestamp
+                obsval = weeutil.weeutil.to_int(dt.timestamp())
+            # WeeWX value with group?
+            elif obsweewx[2] is not None:
+                obsval = weeutil.weeutil.to_float(obsval)
+            y[obsweewx[0]] = (obsval, obsweewx[1], obsweewx[2])
+
+        # get primary source data
+        source_id = weather.get('source_id')
+        if source_id is not None:
+            for source in sources:
+                if source.get('id') == source_id:
+                    for obsapi, obsweewx in self.sources_obs.items():
+                        obsname = self.prefix+obsweewx[0][0].upper()+obsweewx[0][1:]
+                        obsval = source.get(obsapi)
+                        if obsval is None:
+                            if self.debug >= 2:
+                                logdbg("thread '%s': No value for source '%s' - '%s'" % (self.name, str(obsapi), str(obsweewx[0])))
+                            continue
+
+                        if self.debug >= 3:
+                            logdbg("thread '%s': weewx=%s api=%s obs=%s val=%s" % (self.name, str(obsweewx[0]), str(obsapi), str(obsname), str(obsval)))
+
+                        # WeeWX value with group?
+                        if obsweewx[2] is not None:
+                            obsval = weeutil.weeutil.to_float(obsval)
+                        
+                        y[obsweewx[0]] = (obsval, obsweewx[1], obsweewx[2])
+                    break
+
+        # convert Brightsky condition to weather description and icon
+        apiicon = weather.get('icon')
+        if apiicon is None:
+            # dummy
+            apiicon = 'unknown'
+        condition = self.get_conditions(apiicon)
+
+        if condition is not None:
+            y['icon'] = (condition[self.iconset], None, None)
+            y['icontitle'] = (condition[0], None, None)
+
+        night = is_night(y, log_success=(self.log_success or self.debug > 0),
+                         log_failure=(self.log_failure or self.debug > 0))
+        if night != "N/A":
+            y['day'] = (0 if night else 1,'count','group_count')
+        else:
+            y['day'] = (None,'count','group_count')
+
+        x.append(y)
+
+        if self.debug > 0:
+            logdbg("thread '%s': result=%s" % (self.name, str(x)))
+
+        try:
+            self.lock.acquire()
+            self.data = x
+        finally:
+            self.lock.release()
+
+
+
+# ============================================================================
+#
+# Class CurrentService
+#
+# ============================================================================
+
+class CurrentService(StdService):
 
     def __init__(self, engine, config_dict):
-        super(DWDservice,self).__init__(engine, config_dict)
+        super(CurrentService,self).__init__(engine, config_dict)
```

Comment:
Der Service soll perspektivisch auch die anderen Teile laden. Außerdem muß man genau überlegen, ob man etwas ändert, daß bei einem Update einer bestehenden Installation dazu führt, daß es auf einmal nicht mehr geht. Und das ist die Folge, wenn man DWDservice umbenennt.

Reply:
Ich habe das gemacht, weil Du `dwd.py` durch `weatherservices.py` ersetzt hast. Daher auch der "harte Schnitt" in der Konfiguration und in der Benennung des Services. Wobei Deine Hinweise auch zutreffen.

Reply:
Habe ich das?

Wie auch immer, den "harten Schnitt" können wir uns für später aufheben. Jetzt ist es dafür zu früh.

Reply:
> Habe ich das?

Ja ;-)

> Wie auch immer, den "harten Schnitt" können wir uns für später aufheben. Jetzt ist es dafür zu früh.

passe ich an

Da wäre es vorteilhaft gewesen, `dwd.py` nicht zu ersetzen sondern auf `weatherservices.py` zu verweisen. `dwd.py` hätte die bis dahin bestehende Konfiguration genutzt (POI, CDC, ZAMG, Open-Meteo) und `weatherservices.py` konsequent eine  entsprechend angepasste.